From 60eee2e81d3ba7825dc49aa62223f6f134a912b4 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 25 May 2021 17:43:50 +0300
Subject: [PATCH 1544/1921] marvell: phy-mvebu-cp110-comphy: Add power ON
 support for 2500BASET mode

Add PHY_INTERFACE_MODE_2500BASET to mvebu_comphy_power_on callback.

Change-Id: I305f1ae4e74765bdebc78b4385998775509477f4
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52606
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
(cherry picked from commit 4430734cbc05e576e1bc5a0c37cedfc89d736997)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52707
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/phy/marvell/phy-mvebu-cp110-comphy.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
index b6dec6970ab0..1dcbc75d628c 100644
--- a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
+++ b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
@@ -797,7 +797,8 @@ static int mvebu_comphy_power_on(struct phy *phy)
 			fw_speed = COMPHY_FW_SPEED_1250;
 			break;
 		case PHY_INTERFACE_MODE_2500BASEX:
-			dev_dbg(priv->dev, "set lane %d to 2500BASE-X mode\n",
+		case PHY_INTERFACE_MODE_2500BASET:
+			dev_dbg(priv->dev, "set lane %d to 2500BASE mode\n",
 				lane->id);
 			fw_speed = COMPHY_FW_SPEED_3125;
 			break;
-- 
2.31.1

